---
title: "full_profile_changes_fiber_vs_fermented"
author: "HCW"
date: "2/7/2020"
output: html_document
---

Save paths and formulas
```{r}
save_path_diet <- "../../data/diet/cleaned/"
save_path_16s <- "../../data/16S/"
save_path_immune <- "../../data/combined_immune/"
save_path_metagenomics <- "../../data/metagenomics/cleaned/"
save_path_scfa <- "../../data/scfa/"
save_path_proteomics <- "../../data/proteomics/cleaned/"
save_path_metagenomics <- "../../data/metagenomics/cleaned/"

save_path_fvf <- "../../data/full_profile_changes_fiber_fermented/"

save_path_figures <- "../../plots/"

loocv_rf_function <- function(df) {
  set.seed(20)
  folds <- groupKFold(df$Participant,k=length(unique(df$Participant)))
  control <- rfeControl(functions=rfFuncs, method="LOOCV",number = 100,index=folds)
  df_temp <- select(df,-Participant,-Timepoint) %>% as.data.frame
  df_centered <- center_scale(df_temp[,2:dim(df_temp)[2]])
  rfe_obj <- rfe(df_centered,as.factor(df_temp[,1]), sizes=(2:ncol(df_temp)-1), rfeControl=control)
  return(rfe_obj)
}
center_scale <- function(x) {
    scale(x, scale = FALSE)
}

diff_from_1 <- function(df_gather){
  df_gather_1 <- df_gather %>% 
    filter(Timepoint==1) %>% 
    dplyr::rename(Result_base=Result) %>% 
    select(-Timepoint)
  df_gather_not_1 <- df_gather %>% 
    filter(Timepoint!=1)
  df_gather_diff <- left_join(df_gather_1,df_gather_not_1) %>% 
    mutate(Result_diff=Result-Result_base) %>% 
    select(-Result_base,-Result) 
  return(df_gather_diff)
}

loocv_results_dfs <- function(loocv_object,data_type_char){
  accuracy_df <- data.frame(data_type=data_type_char,
                            max_acc=max(loocv_object[["results"]][["Accuracy"]]),
                            optsize=loocv_object[["optsize"]])
  optVariables_df <- data.frame(data_type=data_type_char,
                            rank_order=c(1:length(loocv_object[["optVariables"]])),
                            optVariables=loocv_object[["optVariables"]])
  return(list(accuracy_df,optVariables_df))
}


```

Nutrition intake 
  - 91% accuracy
  - 3 parameters: animal protein, total fiber, insoluble dietary fiber 
```{r}
daily_totals_averaged <- read_csv("~/R/fiber-fermented-study/data/diet/raw/pilot_NDSR_daily_totals_averaged.csv") %>% 
  dplyr::rename(Participant="Record ID")
names(daily_totals_averaged) <- make.names(colnames(daily_totals_averaged))

diet_key <- read_csv("~/R/fiber-fermented-study/data/metadata/diet_key.csv")

params_to_test <- c("Energy (kcal)",
                    "Total Fat (g)" ,                                                 
                    "Total Carbohydrate (g)",                                                
                   "Total Protein (g)",                                              
                   "Animal Protein (g)" ,                                             
                   "Vegetable Protein (g)",
                   "Cholesterol (mg)",                                                        
                   "Total Saturated Fatty Acids (SFA) (g)",                                            
                   "Total Monounsaturated Fatty Acids (MUFA) (g)",                                     
                   "Total Polyunsaturated Fatty Acids (PUFA) (g)" ,   
                  "Total Dietary Fiber (g)",                                  
                   "Soluble Dietary Fiber (g)",                                  
                   "Insoluble Dietary Fiber (g)" ,                                 
                   "Pectins (g)",
                  "Sodium (mg)",
                  "Total Sugars (g)",  
                  "Added Sugars (by Total Sugars) (g)",                                               
                   "Total Grains (ounce equivalents)"  ,                                               
                   "Whole Grains (ounce equivalents)"   ,                                              
                   "Refined Grains (ounce equivalents)",
                  "Beta-Carotene (provitamin A carotenoid) (mcg)",
                  "Beta-Cryptoxanthin (provitamin A carotenoid) (mcg)",
                  "Alpha-Carotene (provitamin A carotenoid) (mcg)",
                  "Lutein + Zeaxanthin (mcg)",
                  "Lycopene (mcg)",
                  "Magnesium (mg)",
                  "Vitamin C (ascorbic acid) (mg)",
                  "Vitamin K (phylloquinone) (mcg)",
                  "Potassium (mg)",
                  "Calcium (mg)",
                  "Iron (mg)") %>% 
  make.names()

daily_totals_averaged$Timepoint <- recode_factor(daily_totals_averaged$Visit.Number,
                                                 "BL"=1,
                                                 "W4"=4,
                                                 "W10"=7,
                                                 "W14"=9)

nutrients_data <- daily_totals_averaged %>% 
  left_join(.,diet_key) %>% 
  select(`Participant`, `Timepoint`, `Group`, !!!params_to_test) %>% 
  na.omit() %>% 
  filter(! Participant %in% c(8000,8012))

nutrients_data_gather <- nutrients_data %>% 
  gather(key=Analyte,value=Result,-c(Participant:Group)) 
nutrients_data_diff <- diff_from_1(nutrients_data_gather)

nutrients_data_diff_tp7 <- nutrients_data_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff) 
loocv_rf_nutrients <- nutrients_data_diff_tp7 %>% 
  loocv_rf_function()
loocv_rf_results_nutrients <- loocv_rf_nutrients %>% 
  loocv_results_dfs(.,"diet_nutrient_intake")
saveRDS(loocv_rf_results_nutrients,paste(save_path_fvf,"loocv_rf_results_nutrients.rds",sep=""))

#plot significant differences
sigg_loocv_rf_nutrients_diff_tp7 <- nutrients_data_diff %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_nutrients[[2]]$optVariables)
ggplot(sigg_loocv_rf_nutrients_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_violin(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Diet Group")+
  ylab("Nutrient intake difference from Week 10 to Week 0")+
  theme(legend.position="none")+
  ggtitle("Dietary nutrient intake")
ggsave(paste(save_path_figures,"loocv_rf_nutrients_diff.pdf"),width = 5,height=4,useDingbats=FALSE)

```

Immune status parameters (olink, cytof, phosphoflow)
```{r}
#Immune difference data
immune_diff_all <- read.csv(paste(save_path_immune,"combined_immune_feature_differences_set.csv",sep=""))
immune_diff_data_key <- read.csv(paste(save_path_immune,"immune_feature_key.csv",sep=""))

#filter feature key to get names for data frame filtering
feature_ids_tp6 <- make.names(filter(immune_diff_data_key,Time=="T6")$Feature) 
feature_ids_tp6_cytokine <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Cytokine")$Feature)
feature_ids_tp6_cell_freq <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Cell frequencies")$Feature)
feature_ids_tp6_endog_sig <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Endogenous signaling")$Feature)
feature_ids_tp6_sig_cap <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Signaling capacity")$Feature)

#filter immune differences data frame 
immune_diff_tp6 <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6)))
immune_diff_tp6_cytokine <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_cytokine)))
immune_diff_tp6_cell_freq <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_cell_freq)))
immune_diff_tp6_endog_sig <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_endog_sig)))
immune_diff_tp6_sig_cap <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_sig_cap)))

loocv_rf_immune_cytokines <- immune_diff_tp6_cytokine %>% 
  mutate(Timepoint=6) %>% 
  loocv_rf_function()
loocv_rf_results_cytokines <- loocv_rf_immune_cytokines %>% 
  loocv_results_dfs(.,"cytokine")
saveRDS(loocv_rf_results_cytokines,paste(save_path_fvf,"loocv_rf_results_cytokines.rds",sep=""))

loocv_rf_immune_cell_freq <- immune_diff_tp6_cell_freq %>% 
  mutate(Timepoint=6) %>% 
  loocv_rf_function()
loocv_rf_results_cell_freq <- loocv_rf_immune_cell_freq %>% 
  loocv_results_dfs(.,"immune_cell_frequency")
saveRDS(loocv_rf_results_cell_freq,paste(save_path_fvf,"loocv_rf_results_cell_freq.rds",sep=""))

loocv_rf_immune_endog_sig <- immune_diff_tp6_endog_sig %>% 
  mutate(Timepoint=6) %>% 
  loocv_rf_function()
loocv_rf_results_endog_sig <- loocv_rf_immune_endog_sig %>% 
  loocv_results_dfs(.,"immune_endogenous_signaling")
saveRDS(loocv_rf_results_endog_sig,paste(save_path_fvf,"loocv_rf_results_endog_sig.rds",sep=""))

loocv_rf_immune_sig_cap <- immune_diff_tp6_sig_cap %>% 
  mutate(Timepoint=6) %>% 
  na.omit() %>% 
  loocv_rf_function()
loocv_rf_results_sig_cap <- loocv_rf_immune_sig_cap %>% 
  loocv_results_dfs(.,"immune_signaling_capacity")
saveRDS(loocv_rf_results_sig_cap,paste(save_path_fvf,"loocv_rf_results_sig_cap.rds",sep=""))

```

ASV rank list 
  - 80% accuracy
  - 1 parameter, lachnospira 
```{r}
#filtered ASV rankings 
asv_rank_data_spread <- readRDS(file = paste(save_path_16s, "asv_rank_fiber_fermented.rds", sep = "")) %>% 
  dplyr::rename(Analyte=ASV,Result=rank_value) %>% 
  mutate(data_type="ASV_rank") %>% 
  spread(key=Analyte,value=Result)
taxa.raw <- select(asv_rank_data_spread, -c(Participant:data_type))
filter.index1 <- apply(taxa.raw,2,function(X){sum(X>1)>0.25*length(X)}) #filter asvs to at least 25% of samples
taxa.filter <- taxa.raw[,filter.index1]
asv_filt_rank_data <- select(asv_rank_data_spread, c(Participant:data_type)) %>%
  bind_cols(., taxa.filter) %>% 
  gather(key=Analyte,value=Result,-c(Participant:data_type))
asv_filt_rank_diff <- asv_filt_rank_data %>% 
  select(-data_type) %>% 
  diff_from_1()

asv_filt_rank_diff_tp7 <- asv_filt_rank_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff) 
loocv_rf_asv_filt <- asv_filt_rank_diff_tp7%>% 
  loocv_rf_function()
loocv_rf_results_asv <- loocv_rf_asv_filt %>% 
  loocv_results_dfs(.,"asv_filt_rank")
saveRDS(loocv_rf_results_asv,paste(save_path_fvf,"loocv_rf_results_asv.rds",sep=""))

sigg_loocv_rf_asv_diff_tp7 <- asv_filt_rank_diff %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_asv[[2]]$optVariables)
ggplot(sigg_loocv_rf_asv_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_violin(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Diet Group")+
  ylab("ASV rank order difference from Week 10 to Week 0")+
  theme(legend.position="none")+
  ggtitle("16S ASVs")
ggsave(paste(save_path_figures,"loocv_rf_asv_diff.pdf"),width = 3,height=4,useDingbats=FALSE)

#find taxa of the asv variable
ps_dada2_with_tree <- readRDS( file = paste(save_path_16s, "phyloseq_obj_PilotStudy_tree_redo_fwdONLY.rds", sep = ""))
asv_taxa_tbl <- data.frame(ps_dada2_with_tree@tax_table@.Data) %>% 
  mutate(ASV=rownames(.))
sigg_asv_taxa_tbl <- asv_taxa_tbl %>% 
  filter(ASV %in% loocv_rf_results_asv[[2]]$optVariables) #lachnospira
```

Alpha diversity - observed and shannon
  - 60% accuracy 
```{r}
#alpha diversity, observed and shannon measures
alpha_diversity_data <- readRDS(file = paste(save_path_16s, "samdf_diversity.rds", sep = "")) %>% 
  gather(key=Analyte,value=Result,Observed,Shannon) %>% 
  select(Participant,Timepoint,Group,Analyte,Result)
alpha_diversity_diff <- alpha_diversity_data %>% 
  diff_from_1()
alpha_diversity_diff_tp7 <- alpha_diversity_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff) 
loocv_rf_alpha_diversity <- alpha_diversity_diff_tp7 %>% 
  loocv_rf_function()
loocv_rf_results_alpha_diversity <- loocv_rf_alpha_diversity %>% 
  loocv_results_dfs(.,"alpha_diversity")
saveRDS(loocv_rf_results_alpha_diversity,paste(save_path_fvf,"loocv_rf_results_alpha_diversity.rds",sep=""))
```

SCFA data
  - 66% accuracy
  - 1 parameter, valeric acid 
```{r}
#SCFA data, stool samples
scfa_data <- readRDS(file = paste(save_path_scfa, "scfa_data.rds", sep = "")) %>% 
  select(Participant,Timepoint,Group,Analyte,Result) %>% 
  filter(Analyte != "Lactic acid")
scfa_data$Analyte %<>% make.names
scfa_diff <- scfa_data %>% 
  diff_from_1()
scfa_diff_tp7 <- scfa_diff %>% 
  filter(Timepoint %in% c(7)) %>% 
  spread(key=Analyte,value=Result_diff) 
loocv_rf_scfa <- scfa_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_scfa <- loocv_rf_scfa %>% 
  loocv_results_dfs(.,"scfa")
saveRDS(loocv_rf_results_scfa,paste(save_path_fvf,"loocv_rf_results_scfa.rds",sep=""))
```

Proteomics, host data
```{r}
proteo_host_diff_filt <- readRDS(file = paste(save_path_proteomics, "proteo_host_diff_filt.rds", sep = ""))
proteo_host_diff_tp7 <- proteo_host_diff_filt %>% 
  filter(Timepoint==7)
loocv_rf_proteo_host <- proteo_host_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_proteo_host <- loocv_rf_proteo_host %>% 
  loocv_results_dfs(.,"proteo_host") 
saveRDS(loocv_rf_results_proteo_host,paste(save_path_fvf,"loocv_rf_results_proteo_host.rds",sep=""))

#plot differences in fiber vs. fermented loocv rf
sigg_loocv_rf_proteo_host_diff_tp7 <- proteo_host_diff_filt %>% 
  gather(key=Analyte,value=Result_diff,-Participant,-Group,-Timepoint) %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_proteo_host[[2]]$optVariables)
ggplot(sigg_loocv_rf_proteo_host_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_boxplot(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Diet Group")+
  ylab("Log2 difference from Week 10 to Week 0")+
  theme(legend.position="none")+
  ggtitle("Host proteomics")
ggsave(paste(save_path_figures,"loocv_rf_proteo_host.pdf"),width = 2,height=4,useDingbats=FALSE)

#find description of sigg host proteins
proteo_host_gather_norm <- readRDS(file = paste(save_path_proteomics, "proteo_host_gather_norm.rds", sep = ""))
proteo_host_description_key <- proteo_host_gather_norm %>% 
  select(Accession,Description)

sigg_proteo_host_description_key <- proteo_host_description_key %>% 
  filter(Accession %in% loocv_rf_results_proteo_host[[2]]$optVariables)
```

Proteomics, microbe data
```{r}
proteo_microbe_desc_diff_filt <- readRDS(file = paste(save_path_proteomics, "proteo_microbe_desc_diff_filt.rds", sep = ""))
proteo_microbe_diff_tp7 <- proteo_microbe_desc_diff_filt %>% 
  filter(Timepoint==7)
loocv_rf_proteo_microbe <- proteo_microbe_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_proteo_microbe <- loocv_rf_proteo_microbe %>% 
  loocv_results_dfs(.,"proteo_microbe") 
saveRDS(loocv_rf_results_proteo_microbe,paste(save_path_fvf,"loocv_rf_results_proteo_microbe.rds",sep=""))

#plot differences in fiber vs. fermented loocv rf
sigg_loocv_rf_proteo_microbe_diff_tp7 <- proteo_microbe_desc_diff_filt %>% 
  gather(key=Analyte,value=Result_diff,-Participant,-Group,-Timepoint) %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_proteo_microbe[[2]]$optVariables)
ggplot(sigg_loocv_rf_proteo_microbe_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_violin(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Microbe Protein")+
  ylab("Log2 fold change difference from Week 10 to Week 0")+
  theme(legend.position="none")

#find description of the microbe proteins
proteo_microbe_sum_desc_gather_norm <- readRDS(file = paste(save_path_proteomics, "proteo_microbe_sum_desc_gather_norm.rds", sep = ""))
microbe_description_key <- proteo_microbe_sum_desc_gather_norm %>% 
  ungroup() %>% 
  select(Description,Description_ID) %>% 
  unique

sigg_microbe_description_key <- microbe_description_key %>% 
  filter(Description_ID %in% loocv_rf_results_proteo_microbe[[2]]$optVariables)
```

CAzyme data
```{r}
#cazyme analysis data 
cazyme_data <- readRDS(paste(save_path_metagenomics,"cazyme_filter_all_gather.rds")) %>%
  mutate(Analyte=paste(CAzyme_annot,CAzyme,sep="_")) %>% 
  select(Participant,Timepoint,Group,Analyte,rpm) %>% 
  dplyr::rename(Result=rpm) 
cazyme_diff <- cazyme_data %>% 
  diff_from_1()
cazyme_diff_tp7 <- cazyme_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff)
loocv_rf_cazymes <- cazyme_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_cazymes <- loocv_rf_cazymes %>% 
  loocv_results_dfs(.,"cazymes")
saveRDS(loocv_rf_results_cazymes,paste(save_path_fvf,"loocv_rf_results_cazymes.rds",sep=""))
```

```{r}
#concat microbiome loocv results 
microbiome_loocv_results = data.frame(bind_rows(loocv_rf_results_alpha_diversity[[1]],
                                                loocv_rf_results_asv[[1]],
                                                loocv_rf_results_scfa[[1]],
                                                loocv_rf_results_proteo_microbe[[1]],
                                                loocv_rf_results_cazymes[[1]]))
microbiome_optVariables = data.frame(bind_rows(loocv_rf_results_alpha_diversity[[2]],
                                                loocv_rf_results_asv[[2]],
                                                loocv_rf_results_scfa[[2]],
                                                loocv_rf_results_proteo_microbe[[2]],
                                                loocv_rf_results_cazymes[[2]]))

#concat host loocv results
host_loocv_results = data.frame(bind_rows(loocv_rf_results_cytokines[[1]],
                                          loocv_rf_results_cell_freq[[1]],
                                          loocv_rf_results_endog_sig[[1]],
                                          loocv_rf_results_sig_cap[[1]],
                                          loocv_rf_results_proteo_host[[1]]))
host_optVariables = data.frame(bind_rows(loocv_rf_results_cytokines[[2]],
                                          loocv_rf_results_cell_freq[[2]],
                                          loocv_rf_results_endog_sig[[2]],
                                          loocv_rf_results_sig_cap[[2]],
                                          loocv_rf_results_proteo_host[[2]]))

#diet results (nutritional intake)
diet_loocv_results = data.frame(loocv_rf_results_nutrients[[1]])
diet_optVariables = data.frame(loocv_rf_results_nutrients[[2]])
```

Plot accuracy from the diet, microbe and host models, individual platforms
```{r}
loocv_results_all_platforms <- bind_rows(data.frame(system_studied="Microbe",microbiome_loocv_results),
                                         data.frame(system_studied="Host",host_loocv_results),
                                         data.frame(system_studied="Diet",diet_loocv_results)) 
ggplot(loocv_results_all_platforms,aes(x=reorder(data_type,max_acc),y=max_acc,fill=system_studied))+
  geom_bar(stat="identity",color="#525252")+
  scale_fill_manual(breaks=c("Microbe","Host","Diet"),values=c("#636363","#f0f0f0","#bdbdbd"))+
  theme_bw()+
  geom_hline(yintercept=0.5,linetype="dashed",color="#525252")+
  ylab("LOOCV RF max accuracy")+
  ylim(0,1)+
  coord_flip()
ggsave(paste(save_path_figures,"fiber_fermented_loocv_rf_microbe_features.pdf"),width = 7.6,height=7,useDingbats=FALSE)
```

Pairwise comparison of dendrograms
```{r}
ecdf_fun <- function(x,perc) ecdf(x)(perc)
ecdf_fun(1:10,8)

compare_dend_function <- function(first_char,second_char,df_1,df_2){
  ppt_list <- intersect(df_1$Participant,df_2$Participant)
  
  df_1_dend <- df_1 %>% 
    filter(Participant %in% ppt_list) %>% 
    arrange(Participant) %>% 
    select(-Participant,-Timepoint,-Group) %>% 
    # na.omit() %>% 
    dist %>% 
    hclust("centroid") %>% 
    as.dendrogram
  
  df_2_dend <- df_2 %>% 
    filter(Participant %in% ppt_list) %>% 
    arrange(Participant) %>% 
    select(-Participant,-Timepoint,-Group) %>% 
    # na.omit() %>% 
    dist %>% 
    hclust("centroid") %>% 
    as.dendrogram
  
  dend_list_all <- dendlist("a"=df_1_dend,
                            "b"=df_2_dend)
  #cors <- cor.dendlist(dend_list_all)
  cor_coph <- cor_cophenetic(dend_list_all)
  
  R <- 1000
  cor_cophenetic_dist_1 <- numeric(R)
  cor_cophenetic_dist_2 <- numeric(R)
  set.seed(20)
  for(i in 1:R) {
     dend_mixed <- sample.dendrogram(df_1_dend, replace = FALSE)
     dend_mixed_2 <- sample.dendrogram(df_2_dend, replace = FALSE)
     cor_cophenetic_dist_1[i] <- cor_cophenetic(dend_mixed, df_2_dend)
     cor_cophenetic_dist_2[i] <- cor_cophenetic(df_1_dend, dend_mixed_2)
     cor_cophenetic_dist <- c(cor_cophenetic_dist_1,cor_cophenetic_dist_2)
  }
  sigg_cor_coph_95 <- quantile(cor_cophenetic_dist, prob=0.95)
  sigg_cor_coph_99 <- quantile(cor_cophenetic_dist, prob=0.99)
  exact_pvalue <- 1-ecdf_fun(cor_cophenetic_dist,cor_coph)
  return(data.frame(first_compare=first_char,
                    second_compare=second_char,
                    cor_coph=cor_coph,
                    sigg_cor_coph_95=sigg_cor_coph_95,
                    sigg_cor_coph_99=sigg_cor_coph_99,
                    sigg=ifelse(cor_coph >= sigg_cor_coph_99,"**",ifelse(cor_coph>=sigg_cor_coph_95,"*","")),
                    pvalue_exact=exact_pvalue))
}

exp_data_list <- list(nutrients_data_diff_tp7,
                      mutate(immune_diff_tp6_cytokine,Timepoint=6),                      
                      mutate(immune_diff_tp6_endog_sig,Timepoint=6),
                      mutate(immune_diff_tp6_cell_freq,Timepoint=6),                      
                      mutate(immune_diff_tp6_sig_cap,Timepoint=6),                      
                      proteo_host_diff_tp7,                      
                      asv_filt_rank_diff_tp7,                      
                      alpha_diversity_diff_tp7,
                      scfa_diff_tp7,
                      cazyme_diff_tp7,
                      proteo_microbe_diff_tp7)
exp_list_names <- c("Nutrients","Cytokines","Endogenous_signaling","Cell_frequency","Signaling_capacity","Host_proteomics","ASV_rank","Alpha_diversity","SCFAs","CAzymes","Microbe_proteomics")


compare_dend_df <- data.frame(first_compare=c(),
                    second_compare=c(),
                    cor_coph=c(),
                    sigg_cor_coph_05=c(),
                    sigg_cor_coph_95=c(),
                    sigg=c(),
                    pvalue_exact=c())
for(j in 1:length(exp_data_list)) {
  for(i in 1:length(exp_data_list)){
    compare_dend_temp <- compare_dend_function(exp_list_names[[j]],
                                               exp_list_names[[i]],
                                               exp_data_list[[j]],
                                               exp_data_list[[i]])
    compare_dend_df <- bind_rows(compare_dend_df,compare_dend_temp)
    }
}

dat=select(compare_dend_df,first_compare,second_compare)
dat.sort = t(apply(dat, 1, sort))
dat <- dat[!duplicated(dat.sort),]

compare_dend_pvalue_adj <- left_join(dat,compare_dend_df) %>% 
  filter(first_compare!=second_compare)

compare_dend_corr_mat <- compare_dend_df %>% 
  select(first_compare,second_compare,cor_coph) %>% 
  pivot_wider(names_from =second_compare,values_from=cor_coph) 
rownames(compare_dend_corr_mat) <- compare_dend_corr_mat$first_compare
compare_dend_corr_mat <- compare_dend_corr_mat %>% 
  select(-first_compare) %>% 
  as.matrix

corrplot(compare_dend_corr_mat, 
         "circle", 
         "lower",
         diag=FALSE,
         tl.col="black",
         is.corr=FALSE,
         col=colorRampPalette(c("grey","white","blue"))(200))
```

Show one dendrogram tangle 
```{r}
ppt_list <- intersect(cazyme_diff_tp7$Participant,asv_filt_rank_diff_tp7$Participant)

df_1_dend <- cazyme_diff_tp7 %>% 
  filter(Participant %in% ppt_list) %>% 
  arrange(Participant) %>% 
  select(-Participant,-Timepoint,-Group) %>% 
  # na.omit() %>% 
  dist %>% 
  hclust("centroid") %>% 
  as.dendrogram
df_2_dend <- asv_filt_rank_diff_tp7 %>% 
  filter(Participant %in% ppt_list) %>% 
  arrange(Participant) %>% 
  select(-Participant,-Timepoint,-Group) %>% 
  # na.omit() %>% 
  dist %>% 
  hclust("centroid") %>% 
  as.dendrogram

dendlist(df_1_dend, df_2_dend) %>%
  untangle(method = "step1side") %>% # Find the best alignment layout
  tanglegram() 
```


