---
title: "full_profile_changes_fiber_vs_fermented"
author: "HCW"
date: "2/7/2020"
output: html_document
---

```{r}
library(caret)
```


Save paths and formulas
```{r}
save_path_diet <- "../../data/diet/cleaned/"
save_path_16s <- "../../data/16S/"
save_path_immune <- "../../data/combined_immune/"
save_path_metagenomics <- "../../data/metagenomics/cleaned/"
save_path_scfa <- "../../data/scfa/"
save_path_proteomics <- "../../data/proteomics/cleaned/"
save_path_metagenomics <- "../../data/metagenomics/cleaned/"

save_path_fvf <- "../../data/full_profile_changes_fiber_fermented/"

save_path_figures <- "../../plots/"

loocv_rf_function <- function(df) {
  set.seed(20)
  folds <- groupKFold(df$Participant,k=length(unique(df$Participant)))
  control <- rfeControl(functions=rfFuncs, method="LOOCV",number = 100,index=folds)
  df_temp <- select(df,-Participant,-Timepoint) %>% as.data.frame
  df_centered <- center_scale(df_temp[,2:dim(df_temp)[2]])
  rfe_obj <- rfe(df_centered,as.factor(df_temp[,1]), sizes=(2:ncol(df_temp)-1), rfeControl=control)
  return(rfe_obj)
}
center_scale <- function(x) {
    scale(x, scale = FALSE)
}

diff_from_1 <- function(df_gather){
  df_gather_1 <- df_gather %>% 
    filter(Timepoint==1) %>% 
    dplyr::rename(Result_base=Result) %>% 
    select(-Timepoint)
  df_gather_not_1 <- df_gather %>% 
    filter(Timepoint!=1)
  df_gather_diff <- left_join(df_gather_1,df_gather_not_1) %>% 
    mutate(Result_diff=Result-Result_base) %>% 
    select(-Result_base,-Result) 
  return(df_gather_diff)
}

loocv_results_dfs <- function(loocv_object,data_type_char){
  accuracy_df <- data.frame(data_type=data_type_char,
                            max_acc=max(loocv_object[["results"]][["Accuracy"]]),
                            optsize=loocv_object[["optsize"]])
  optVariables_df <- data.frame(data_type=data_type_char,
                            rank_order=c(1:length(loocv_object[["optVariables"]])),
                            optVariables=loocv_object[["optVariables"]])
  return(list(accuracy_df,optVariables_df))
}


```

Nutrition intake 
  - 91% accuracy
  - 3 parameters: animal protein, total fiber, insoluble dietary fiber 
```{r}
daily_totals_averaged <- read_csv("~/R/fiber-fermented-study/data/diet/raw/pilot_NDSR_daily_totals_averaged.csv") %>% 
  dplyr::rename(Participant="Record ID")
names(daily_totals_averaged) <- make.names(colnames(daily_totals_averaged))

diet_key <- read_csv("~/R/fiber-fermented-study/data/metadata/diet_key.csv")

params_to_test <- c("Energy (kcal)",
                    "Total Fat (g)" ,                                                 
                    "Total Carbohydrate (g)",                                                
                   "Total Protein (g)",                                              
                   "Animal Protein (g)" ,                                             
                   "Vegetable Protein (g)",
                   "Cholesterol (mg)",                                                        
                   "Total Saturated Fatty Acids (SFA) (g)",                                            
                   "Total Monounsaturated Fatty Acids (MUFA) (g)",                                     
                   "Total Polyunsaturated Fatty Acids (PUFA) (g)" ,   
                  "Total Dietary Fiber (g)",                                  
                   "Soluble Dietary Fiber (g)",                                  
                   "Insoluble Dietary Fiber (g)" ,                                 
                   "Pectins (g)",
                  "Sodium (mg)",
                  "Total Sugars (g)",  
                  "Added Sugars (by Total Sugars) (g)",                                               
                   "Total Grains (ounce equivalents)"  ,                                               
                   "Whole Grains (ounce equivalents)"   ,                                              
                   "Refined Grains (ounce equivalents)",
                  "Beta-Carotene (provitamin A carotenoid) (mcg)",
                  "Beta-Cryptoxanthin (provitamin A carotenoid) (mcg)",
                  "Alpha-Carotene (provitamin A carotenoid) (mcg)",
                  "Lutein + Zeaxanthin (mcg)",
                  "Lycopene (mcg)",
                  "Magnesium (mg)",
                  "Vitamin C (ascorbic acid) (mg)",
                  "Vitamin K (phylloquinone) (mcg)",
                  "Potassium (mg)",
                  "Calcium (mg)",
                  "Iron (mg)") %>% 
  make.names()

daily_totals_averaged$Timepoint <- recode_factor(daily_totals_averaged$Visit.Number,
                                                 "BL"=1,
                                                 "W4"=4,
                                                 "W10"=7,
                                                 "W14"=9)

nutrients_data <- daily_totals_averaged %>% 
  left_join(.,diet_key) %>% 
  select(`Participant`, `Timepoint`, `Group`, !!!params_to_test) %>% 
  na.omit() %>% 
  filter(! Participant %in% c(8000,8012))

nutrients_data_gather <- nutrients_data %>% 
  gather(key=Analyte,value=Result,-c(Participant:Group)) 
nutrients_data_diff <- diff_from_1(nutrients_data_gather)

nutrients_data_diff_tp7 <- nutrients_data_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff) 
saveRDS(nutrients_data_diff_tp7,paste(save_path_fvf,"nutrients_data_diff_tp7.rds",sep=""))

loocv_rf_nutrients <- nutrients_data_diff_tp7 %>% 
  loocv_rf_function()
loocv_rf_results_nutrients <- loocv_rf_nutrients %>% 
  loocv_results_dfs(.,"diet_nutrient_intake")
saveRDS(loocv_rf_results_nutrients,paste(save_path_fvf,"loocv_rf_results_nutrients.rds",sep=""))

#plot significant differences
sigg_loocv_rf_nutrients_diff_tp7 <- nutrients_data_diff %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_nutrients[[2]]$optVariables)
ggplot(sigg_loocv_rf_nutrients_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_violin(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Diet Group")+
  ylab("Nutrient intake difference from Week 10 to Week 0")+
  theme(legend.position="none")+
  ggtitle("Dietary nutrient intake")
ggsave(paste(save_path_figures,"loocv_rf_nutrients_diff.pdf"),width = 5,height=4,useDingbats=FALSE)

```

Immune status parameters (olink, cytof, phosphoflow)
```{r}
#Immune difference data
immune_diff_all <- read.csv(paste(save_path_immune,"combined_immune_feature_differences_set.csv",sep=""))
immune_diff_data_key <- read.csv(paste(save_path_immune,"immune_feature_key.csv",sep=""))

#filter feature key to get names for data frame filtering
feature_ids_tp6 <- make.names(filter(immune_diff_data_key,Time=="T6")$Feature) 
feature_ids_tp6_cytokine <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Cytokine")$Feature)
feature_ids_tp6_cell_freq <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Cell frequencies")$Feature)
feature_ids_tp6_endog_sig <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Endogenous signaling")$Feature)
feature_ids_tp6_sig_cap <- make.names(filter(immune_diff_data_key,Time=="T6" & Type=="Signaling capacity")$Feature)

#filter immune differences data frame 
immune_diff_tp6 <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6)))
immune_diff_tp6_cytokine <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_cytokine)))
immune_diff_tp6_cell_freq <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_cell_freq)))
immune_diff_tp6_endog_sig <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_endog_sig)))
immune_diff_tp6_sig_cap <- immune_diff_all %>% 
  select(c("Participant","Group",as.character(feature_ids_tp6_sig_cap)))

saveRDS(immune_diff_tp6_cytokine,paste(save_path_fvf,"immune_diff_tp6_cytokine.rds",sep=""))
saveRDS(immune_diff_tp6_cell_freq,paste(save_path_fvf,"immune_diff_tp6_cell_freq.rds",sep=""))
saveRDS(immune_diff_tp6_endog_sig,paste(save_path_fvf,"immune_diff_tp6_endog_sig.rds",sep=""))
saveRDS(immune_diff_tp6_sig_cap,paste(save_path_fvf,"immune_diff_tp6_sig_cap.rds",sep=""))

loocv_rf_immune_cytokines <- immune_diff_tp6_cytokine %>% 
  mutate(Timepoint=6) %>% 
  loocv_rf_function()
loocv_rf_results_cytokines <- loocv_rf_immune_cytokines %>% 
  loocv_results_dfs(.,"cytokine")
saveRDS(loocv_rf_results_cytokines,paste(save_path_fvf,"loocv_rf_results_cytokines.rds",sep=""))

loocv_rf_immune_cell_freq <- immune_diff_tp6_cell_freq %>% 
  mutate(Timepoint=6) %>% 
  loocv_rf_function()
loocv_rf_results_cell_freq <- loocv_rf_immune_cell_freq %>% 
  loocv_results_dfs(.,"immune_cell_frequency")
saveRDS(loocv_rf_results_cell_freq,paste(save_path_fvf,"loocv_rf_results_cell_freq.rds",sep=""))

loocv_rf_immune_endog_sig <- immune_diff_tp6_endog_sig %>% 
  mutate(Timepoint=6) %>% 
  loocv_rf_function()
loocv_rf_results_endog_sig <- loocv_rf_immune_endog_sig %>% 
  loocv_results_dfs(.,"immune_endogenous_signaling")
saveRDS(loocv_rf_results_endog_sig,paste(save_path_fvf,"loocv_rf_results_endog_sig.rds",sep=""))

loocv_rf_immune_sig_cap <- immune_diff_tp6_sig_cap %>% 
  mutate(Timepoint=6) %>% 
  na.omit() %>% 
  loocv_rf_function()
loocv_rf_results_sig_cap <- loocv_rf_immune_sig_cap %>% 
  loocv_results_dfs(.,"immune_signaling_capacity")
saveRDS(loocv_rf_results_sig_cap,paste(save_path_fvf,"loocv_rf_results_sig_cap.rds",sep=""))

loocv_rf_results_cytokines <- readRDS(paste(save_path_fvf,"loocv_rf_results_cytokines.rds",sep=""))
loocv_rf_immune_cell_freq <- readRDS(paste(save_path_fvf,"loocv_rf_results_cell_freq.rds",sep=""))
loocv_rf_immune_endog_sig <- readRDS(paste(save_path_fvf,"loocv_rf_results_endog_sig.rds",sep=""))
loocv_rf_immune_sig_cap <- readRDS(paste(save_path_fvf,"loocv_rf_results_sig_cap.rds",sep=""))

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
## note: have to import immune data again since big immune feature data frames filtered out difference between baseline timpeoints

## olink
save_path_olink <- "../../data/Olink/cleaned/"
olink_data_gather <- readRDS(paste(save_path_olink, "olink_data_cleaned_gather.rds", sep = ""))
olink_all_diff <- olink_data_gather %>% 
  diff_from_1()
olink_all_diff_tp2 <- olink_all_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff)

loocv_rf_olink_all_tp2 <- olink_all_diff_tp2 %>% 
  loocv_rf_function()
loocv_rf_results_olink_all_tp2 <- loocv_rf_olink_all_tp2 %>% 
  loocv_results_dfs(.,"olink_tp2")
saveRDS(loocv_rf_results_olink_all_tp2, paste(save_path_fvf,"loocv_rf_results_olink_all_tp2.rds",sep=""))

#cytof cell frequency
save_path_cytof_cell_freq <- "../../data/CyTOF/cleaned/"
cell_freq_gather <- readRDS(paste(save_path_cytof_cell_freq, "cytof_cell_freq_gather.rds", sep = ""))
cell_freq_all_diff <- cell_freq_gather %>% 
  diff_from_1()
cell_freq_all_diff_tp2 <- cell_freq_all_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff)

loocv_rf_cell_freq_tp2 <- cell_freq_all_diff_tp2 %>% 
  loocv_rf_function()
loocv_rf_results_cell_freq_tp2 <- loocv_rf_cell_freq_tp2 %>% 
  loocv_results_dfs(.,"cell_freq_tp2")
saveRDS(loocv_rf_results_cell_freq_tp2, paste(save_path_fvf,"loocv_rf_results_cell_freq_tp2.rds",sep=""))

#cytof cell signaling
cell_sig_gather <- readRDS(paste(save_path_cytof_cell_freq, "cytof_signaling_gather.rds", sep = ""))
cell_sig_all_diff <- cell_sig_gather %>% 
  diff_from_1()
cell_sig_all_diff_tp2 <- cell_sig_all_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff)

loocv_rf_cell_sig_tp2 <- cell_sig_all_diff_tp2 %>% 
  loocv_rf_function()
loocv_rf_results_cell_sig_tp2 <- loocv_rf_cell_sig_tp2 %>% 
  loocv_results_dfs(.,"cell_sig_tp2") #77% accuracy
saveRDS(loocv_rf_results_cell_sig_tp2, paste(save_path_fvf,"loocv_rf_results_cell_sig_tp2.rds",sep=""))

### plot differences between groups in tp2
loocv_rf_results_cell_sig_tp2_analytes <- loocv_rf_results_cell_sig_tp2[[2]]$optVariables
cell_sig_all_diff_tp2_loocvRF <- cell_sig_all_diff_tp2 %>% 
  gather(key = Analyte,value=Result_diff,-Participant,-Group,-Timepoint) %>% 
  filter(Analyte %in% loocv_rf_results_cell_sig_tp2_analytes)

ggplot(cell_sig_all_diff_tp2_loocvRF,aes(x=Group,y=Result_diff,fill=Group))+
  geom_boxplot() +
  theme_bw() +
  facet_wrap(~Analyte,scales="free")

#phosphoflow cell signaling capacity
save_path_sig_cap <- "../../data/phosphoflow/cleaned/"
cell_sig_cap_gather <- read.csv(paste(save_path_sig_cap, "phosphoflow_feature_set.csv", sep = "")) %>% 
  select(-DataSet) %>% 
  gather(key=Analyte,value=Result,-Participant,-Timepoint,-Group)
cell_sig_cap_all_diff <- cell_sig_cap_gather %>% 
  diff_from_1()
cell_sig_cap_all_diff_tp2 <- cell_sig_cap_all_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff) %>% 
  na.omit

loocv_rf_cell_sig_cap_tp2 <- cell_sig_cap_all_diff_tp2 %>% 
  loocv_rf_function()
loocv_rf_results_cell_sig_cap_tp2 <- loocv_rf_cell_sig_cap_tp2 %>% 
  loocv_results_dfs(.,"cell_sig_cap_tp2") #65% accuracy
saveRDS(loocv_rf_results_cell_sig_cap_tp2, paste(save_path_fvf,"loocv_rf_results_cell_sig_cap_tp2.rds",sep=""))
```

ASV rank list 
  - 80% accuracy
  - 1 parameter, lachnospira 
```{r}
#filtered ASV rankings 
asv_rank_data_spread <- readRDS(file = paste(save_path_16s, "asv_rank_fiber_fermented.rds", sep = "")) %>% 
  dplyr::rename(Analyte=ASV,Result=rank_value) %>% 
  mutate(data_type="ASV_rank") %>% 
  spread(key=Analyte,value=Result)
taxa.raw <- select(asv_rank_data_spread, -c(Participant:data_type))
filter.index1 <- apply(taxa.raw,2,function(X){sum(X>1)>0.25*length(X)}) #filter asvs to at least 25% of samples
taxa.filter <- taxa.raw[,filter.index1]
asv_filt_rank_data <- select(asv_rank_data_spread, c(Participant:data_type)) %>%
  bind_cols(., taxa.filter) %>% 
  gather(key=Analyte,value=Result,-c(Participant:data_type))
asv_filt_rank_diff <- asv_filt_rank_data %>% 
  select(-data_type) %>% 
  diff_from_1()

asv_filt_rank_diff_tp7 <- asv_filt_rank_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff) 
saveRDS(asv_filt_rank_diff_tp7,paste(save_path_fvf,"asv_filt_rank_diff_tp7.rds",sep=""))

loocv_rf_asv_filt <- asv_filt_rank_diff_tp7%>% 
  loocv_rf_function()
loocv_rf_results_asv <- loocv_rf_asv_filt %>% 
  loocv_results_dfs(.,"asv_filt_rank")
saveRDS(loocv_rf_results_asv,paste(save_path_fvf,"loocv_rf_results_asv.rds",sep=""))
loocv_rf_results_asv <- readRDS(paste(save_path_fvf,"loocv_rf_results_asv.rds",sep=""))


sigg_loocv_rf_asv_diff_tp7 <- asv_filt_rank_diff %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_asv[[2]]$optVariables)
ggplot(sigg_loocv_rf_asv_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_violin(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Diet Group")+
  ylab("ASV rank order difference from Week 10 to Week 0")+
  theme(legend.position="none")+
  ggtitle("16S ASVs")
ggsave(paste(save_path_figures,"loocv_rf_asv_diff.pdf"),width = 3,height=4,useDingbats=FALSE)

#find taxa of the asv variable
ps_dada2_with_tree <- readRDS( file = paste(save_path_16s, "phyloseq_obj_PilotStudy_tree_redo_fwdONLY.rds", sep = ""))
asv_taxa_tbl <- data.frame(ps_dada2_with_tree@tax_table@.Data) %>% 
  mutate(ASV=rownames(.))
sigg_asv_taxa_tbl <- asv_taxa_tbl %>% 
  filter(ASV %in% loocv_rf_results_asv[[2]]$optVariables) #lachnospira

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
asv_filt_rank_diff_tp2 <- asv_filt_rank_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff) 

loocv_rf_asv_filt_tp2 <- asv_filt_rank_diff_tp2%>% 
  loocv_rf_function()
loocv_rf_results_asv_tp2 <- loocv_rf_asv_filt_tp2 %>% 
  loocv_results_dfs(.,"asv_filt_rank") #46% accuracy
saveRDS(loocv_rf_results_asv_tp2, paste(save_path_fvf,"loocv_rf_results_asv_tp2.rds",sep=""))

```

Alpha diversity - observed and shannon
  - 60% accuracy 
```{r}
#alpha diversity, observed and shannon measures
alpha_diversity_data <- readRDS(file = paste(save_path_16s, "samdf_diversity.rds", sep = "")) %>% 
  gather(key=Analyte,value=Result,Observed,Shannon) %>% 
  select(Participant,Timepoint,Group,Analyte,Result)
alpha_diversity_diff <- alpha_diversity_data %>% 
  diff_from_1()
alpha_diversity_diff_tp7 <- alpha_diversity_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff) 
saveRDS(alpha_diversity_diff_tp7,paste(save_path_fvf,"alpha_diversity_diff_tp7.rds",sep=""))

loocv_rf_alpha_diversity <- alpha_diversity_diff_tp7 %>% 
  loocv_rf_function()
loocv_rf_results_alpha_diversity <- loocv_rf_alpha_diversity %>% 
  loocv_results_dfs(.,"alpha_diversity")
saveRDS(loocv_rf_results_alpha_diversity,paste(save_path_fvf,"loocv_rf_results_alpha_diversity.rds",sep=""))
loocv_rf_results_alpha_diversity <- readRDS(paste(save_path_fvf,"loocv_rf_results_alpha_diversity.rds",sep=""))

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
alpha_diversity_diff_tp2 <- alpha_diversity_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff) 

loocv_rf_alpha_diversity_tp2 <- alpha_diversity_diff_tp2 %>% 
  loocv_rf_function()
loocv_rf_results_alpha_diversity_tp2 <- loocv_rf_alpha_diversity_tp2 %>% 
  loocv_results_dfs(.,"asv_filt_rank") #42% accuracy
saveRDS(loocv_rf_results_alpha_diversity_tp2, paste(save_path_fvf,"loocv_rf_results_alpha_diversity_tp2.rds",sep=""))

```

SCFA data
  - 66% accuracy
  - 1 parameter, valeric acid 
```{r}
#SCFA data, stool samples
scfa_data <- readRDS(file = paste(save_path_scfa, "scfa_data.rds", sep = "")) %>% 
  select(Participant,Timepoint,Group,Analyte,Result) %>% 
  filter(Analyte != "Lactic acid")
scfa_data$Analyte %<>% make.names
scfa_diff <- scfa_data %>% 
  diff_from_1()
scfa_diff_tp7 <- scfa_diff %>% 
  filter(Timepoint %in% c(7)) %>% 
  spread(key=Analyte,value=Result_diff) 
saveRDS(scfa_diff_tp7,paste(save_path_fvf,"scfa_diff_tp7.rds",sep=""))

loocv_rf_scfa <- scfa_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_scfa <- loocv_rf_scfa %>% 
  loocv_results_dfs(.,"scfa")
saveRDS(loocv_rf_results_scfa,paste(save_path_fvf,"loocv_rf_results_scfa.rds",sep=""))
loocv_rf_results_scfa <- readRDS(paste(save_path_fvf,"loocv_rf_results_scfa.rds",sep=""))

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
scfa_diff_tp2 <- scfa_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff) 

loocv_rf_scfa_tp2 <- scfa_diff_tp2 %>% 
  loocv_rf_function()
loocv_rf_results_scfa_tp2 <- loocv_rf_scfa_tp2 %>% 
  loocv_results_dfs(.,"asv_filt_rank") #50% accuracy
saveRDS(loocv_rf_results_scfa_tp2, paste(save_path_fvf,"loocv_rf_results_scfa_tp2.rds",sep=""))

```

Proteomics, host data
```{r}
proteo_host_diff_filt <- readRDS(file = paste(save_path_proteomics, "proteo_host_diff_filt.rds", sep = ""))
proteo_host_diff_tp7 <- proteo_host_diff_filt %>% 
  filter(Timepoint==7)
saveRDS(proteo_host_diff_tp7,paste(save_path_fvf,"proteo_host_diff_tp7.rds",sep=""))

#calcualte differences for non-filtered data for IPA data analysis
proteo_host_NOFILT <- readRDS(file = paste(save_path_proteomics, "proteo_host_gather_norm.rds", sep = "")) %>% 
  select(Participant,Group,Timepoint,Accession,avg_log_norm) %>% 
  dplyr::rename(Result=avg_log_norm)
proteo_host_NOFILT_diff <- diff_from_1(proteo_host_NOFILT)
saveRDS(proteo_host_NOFILT_diff,paste(save_path_fvf,"proteo_host_NOFILT_diff.rds",sep=""))


loocv_rf_proteo_host <- proteo_host_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_proteo_host <- loocv_rf_proteo_host %>% 
  loocv_results_dfs(.,"proteo_host") 
saveRDS(loocv_rf_results_proteo_host,paste(save_path_fvf,"loocv_rf_results_proteo_host.rds",sep=""))
loocv_rf_results_proteo_host <- readRDS(paste(save_path_fvf,"loocv_rf_results_proteo_host.rds",sep=""))

#plot differences in fiber vs. fermented loocv rf
sigg_loocv_rf_proteo_host_diff_tp7 <- proteo_host_diff_filt %>% 
  gather(key=Analyte,value=Result_diff,-Participant,-Group,-Timepoint) %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_proteo_host[[2]]$optVariables)
ggplot(sigg_loocv_rf_proteo_host_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_boxplot(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Diet Group")+
  ylab("Log2 difference from Week 10 to Week 0")+
  theme(legend.position="none")+
  ggtitle("Host proteomics")
ggsave(paste(save_path_figures,"loocv_rf_proteo_host.pdf"),width = 2,height=4,useDingbats=FALSE)

#find description of sigg host proteins
proteo_host_gather_norm <- readRDS(file = paste(save_path_proteomics, "proteo_host_gather_norm.rds", sep = ""))
proteo_host_description_key <- proteo_host_gather_norm %>% 
  select(Accession,Description)

sigg_proteo_host_description_key <- proteo_host_description_key %>% 
  filter(Accession %in% loocv_rf_results_proteo_host[[2]]$optVariables)

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
proteo_host_diff_tp2 <- proteo_host_diff_filt %>% 
  filter(Timepoint==2) 

loocv_rf_proteo_host_tp2 <- proteo_host_diff_tp2 %>% 
  loocv_rf_function
loocv_rf_results_proteo_host_tp2 <- loocv_rf_proteo_host_tp2 %>% 
  loocv_results_dfs(.,"proteo_host") # 48% accuracy
saveRDS(loocv_rf_results_proteo_host_tp2, paste(save_path_fvf,"loocv_rf_results_proteo_host_tp2.rds",sep=""))

```

Proteomics, microbe data
```{r}
proteo_microbe_desc_diff_filt <- readRDS(file = paste(save_path_proteomics, "proteo_microbe_desc_diff_filt.rds", sep = ""))
proteo_microbe_diff_tp7 <- proteo_microbe_desc_diff_filt %>% 
  filter(Timepoint==7)
saveRDS(proteo_microbe_diff_tp7,paste(save_path_fvf,"proteo_microbe_diff_tp7.rds",sep=""))


loocv_rf_proteo_microbe <- proteo_microbe_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_proteo_microbe <- loocv_rf_proteo_microbe %>% 
  loocv_results_dfs(.,"proteo_microbe") 
saveRDS(loocv_rf_results_proteo_microbe,paste(save_path_fvf,"loocv_rf_results_proteo_microbe.rds",sep=""))
loocv_rf_results_proteo_microbe <- readRDS(paste(save_path_fvf,"loocv_rf_results_proteo_microbe.rds",sep=""))

#plot differences in fiber vs. fermented loocv rf
sigg_loocv_rf_proteo_microbe_diff_tp7 <- proteo_microbe_desc_diff_filt %>% 
  gather(key=Analyte,value=Result_diff,-Participant,-Group,-Timepoint) %>% 
  filter(Timepoint==7 & Analyte %in% loocv_rf_results_proteo_microbe[[2]]$optVariables)
ggplot(sigg_loocv_rf_proteo_microbe_diff_tp7,aes(x=Analyte,y=Result_diff,fill=Group,color=Group))+
  geom_violin(alpha=0.5,position="dodge")+
  geom_jitter(aes(group=Group), position=position_jitterdodge(), alpha=0.9)+
  scale_fill_manual(values=c('#BEBADA','#99D594'))+
  scale_color_manual(values=c('#BEBADA','#99D594'))+
  theme_bw() +
  xlab("Microbe Protein")+
  ylab("Log2 fold change difference from Week 10 to Week 0")+
  theme(legend.position="none")

#find description of the microbe proteins
proteo_microbe_sum_desc_gather_norm <- readRDS(file = paste(save_path_proteomics, "proteo_microbe_sum_desc_gather_norm.rds", sep = ""))
microbe_description_key <- proteo_microbe_sum_desc_gather_norm %>% 
  ungroup() %>% 
  select(Description,Description_ID) %>% 
  unique

sigg_microbe_description_key <- microbe_description_key %>% 
  filter(Description_ID %in% loocv_rf_results_proteo_microbe[[2]]$optVariables)

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
proteo_microbe_diff_tp2 <- proteo_microbe_desc_diff_filt %>% 
  filter(Timepoint==2) 

loocv_rf_proteo_microbe_tp2 <- proteo_microbe_diff_tp2 %>% 
  loocv_rf_function
loocv_rf_results_proteo_microbe_tp2 <- loocv_rf_proteo_microbe_tp2 %>% 
  loocv_results_dfs(.,"proteo_microbe") # 48% accuracy
saveRDS(loocv_rf_results_proteo_host_tp2, paste(save_path_fvf,"loocv_rf_results_proteo_host_tp2.rds",sep=""))
```

CAzyme data
```{r}
#cazyme analysis data 
cazyme_data <- readRDS(paste(save_path_metagenomics,"cazyme_filter_all_gather.rds")) %>%
  group_by(Participant,Group,Timepoint,CAzyme) %>% 
  dplyr::summarize(rpm_bin_sum=sum(rpm)) %>% 
  dplyr::rename(Analyte=CAzyme,Result=rpm_bin_sum) %>% 
  as.data.frame
  # mutate(Analyte=paste(CAzyme_annot,CAzyme,sep="_")) %>% 
  # select(Participant,Timepoint,Group,Analyte,rpm) %>% 
  # dplyr::rename(Result=rpm) 
cazyme_diff <- cazyme_data %>% 
  diff_from_1()
cazyme_diff_tp7 <- cazyme_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff)
saveRDS(cazyme_diff_tp7,paste(save_path_fvf,"cazyme_diff_tp7.rds",sep=""))

loocv_rf_cazymes <- cazyme_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_cazymes <- loocv_rf_cazymes %>% 
  loocv_results_dfs(.,"cazymes")
saveRDS(loocv_rf_results_cazymes,paste(save_path_fvf,"loocv_rf_results_cazymes.rds",sep=""))
loocv_rf_results_cazymes <- readRDS(paste(save_path_fvf,"loocv_rf_results_cazymes.rds",sep=""))

#cazymes summed bins
cazyme_filter_all_gather_sumbins <- readRDS(paste(save_path_metagenomics,"cazyme_filter_all_gather_sumbins.rds")) %>% 
  ungroup() %>% 
  dplyr::rename(Analyte=CAzyme_bin,Result=sum_rpm)
cazyme_sumbins_diff <- cazyme_filter_all_gather_sumbins %>% 
  diff_from_1()
cazyme_submins_diff_tp7 <- cazyme_sumbins_diff %>% 
  filter(Timepoint==7) %>% 
  spread(key=Analyte,value=Result_diff)
saveRDS(cazyme_submins_diff_tp7,paste(save_path_fvf,"cazyme_sumbins_diff_tp7.rds",sep=""))

loocv_rf_cazymes_sumbins <- cazyme_submins_diff_tp7 %>% 
  loocv_rf_function
loocv_rf_results_cazymes_sumbins <- loocv_rf_cazymes_sumbins %>% 
  loocv_results_dfs(.,"cazymes_sumbins")
saveRDS(loocv_rf_results_cazymes_sumbins,paste(save_path_fvf,"loocv_rf_results_cazymes_sumbins.rds",sep=""))
loocv_rf_results_cazymes_sumbins <- readRDS(paste(save_path_fvf,"loocv_rf_results_cazymes_sumbins.rds",sep=""))

#check loocv rf accuracy predicting diet group based on the difference between the two baseline timepoints (neg control)
cazyme_diff_tp2 <- cazyme_diff %>% 
  filter(Timepoint==2) %>% 
  spread(key=Analyte,value=Result_diff)

loocv_rf_cazymes_tp2 <- cazyme_diff_tp2 %>% 
  loocv_rf_function
loocv_rf_results_cazymes_tp2 <- loocv_rf_cazymes_tp2 %>% 
  loocv_results_dfs(.,"cazymes") # x% accuracy
saveRDS(loocv_rf_results_cazymes_tp2, paste(save_path_fvf,"loocv_rf_results_cazymes_tp2.rds",sep=""))

```

```{r}
#concat microbiome loocv results 
microbiome_loocv_results = data.frame(bind_rows(loocv_rf_results_alpha_diversity[[1]],
                                                loocv_rf_results_asv[[1]],
                                                loocv_rf_results_scfa[[1]],
                                                loocv_rf_results_proteo_microbe[[1]],
                                                loocv_rf_results_cazymes_sumbins[[1]]))
microbiome_optVariables = data.frame(bind_rows(loocv_rf_results_alpha_diversity[[2]],
                                                loocv_rf_results_asv[[2]],
                                                loocv_rf_results_scfa[[2]],
                                                loocv_rf_results_proteo_microbe[[2]],
                                                loocv_rf_results_cazymes_sumbins[[2]]))
saveRDS(microbiome_loocv_results,paste(save_path_fvf,"microbiome_loocv_results.rds",sep=""))
saveRDS(microbiome_optVariables,paste(save_path_fvf,"microbiome_optVariables.rds",sep=""))

#concat host loocv results
host_loocv_results = data.frame(bind_rows(loocv_rf_results_cytokines[[1]],
                                          loocv_rf_immune_cell_freq[[1]],
                                          loocv_rf_immune_endog_sig[[1]],
                                          loocv_rf_immune_sig_cap[[1]],
                                          loocv_rf_results_proteo_host[[1]]))
host_optVariables = data.frame(bind_rows(loocv_rf_results_cytokines[[2]],
                                          loocv_rf_immune_cell_freq[[2]],
                                          loocv_rf_immune_endog_sig[[2]],
                                          loocv_rf_immune_sig_cap[[2]],
                                          loocv_rf_results_proteo_host[[2]]))
saveRDS(host_loocv_results,paste(save_path_fvf,"host_loocv_results.rds",sep=""))
saveRDS(host_optVariables,paste(save_path_fvf,"host_optVariables.rds",sep=""))

#diet results (nutritional intake)
diet_loocv_results = data.frame(loocv_rf_results_nutrients[[1]])
diet_optVariables = data.frame(loocv_rf_results_nutrients[[2]])
```

Plot accuracy from the diet, microbe and host models, individual platforms
```{r}
loocv_results_all_platforms <- bind_rows(data.frame(system_studied="Microbe",microbiome_loocv_results),
                                         data.frame(system_studied="Host",host_loocv_results)) 
ggplot(loocv_results_all_platforms,aes(x=reorder(data_type,max_acc),y=max_acc,fill=system_studied))+
  geom_bar(stat="identity",color="#525252")+
  scale_fill_manual(breaks=c("Microbe","Host","Diet"),values=c("#636363","#f0f0f0","#bdbdbd"))+
  theme_bw()+
  geom_hline(yintercept=0.5,linetype="dashed",color="#525252")+
  ylab("LOOCV RF max accuracy")+
  ylim(0,1)+
  coord_flip()
ggsave(paste(save_path_figures,"fiber_fermented_loocv_rf_microbe_features.pdf"),width = 7.6,height=7,useDingbats=FALSE)

# ggplot(filter(loocv_results_all_platforms,system_studied=="Microbe"),aes(x=reorder(data_type,max_acc),y=max_acc,fill=feature_type))+
#   geom_bar(stat="identity",col="saddlebrown")+
#   scale_fill_manual(breaks=c("individual_features","all_features"),values=c("saddlebrown","white"))+
#   facet_wrap(~system_studied)+
#   theme_bw()+
#   geom_hline(yintercept=0.5,linetype="dashed",color="#525252")+
#   ylab("LOOCV RF max accuracy")+
#   ylim(0,1)+
#   theme(legend.position = "none")+
#   coord_flip()
# ggsave(paste(save_path_figures,"fiber_fermented_loocv_rf_microbe_features.pdf"),width = 4.5,height=5,useDingbats=FALSE)
# 
# ggplot(filter(loocv_results_all_platforms,system_studied=="Host"),aes(x=reorder(data_type, max_acc),y=max_acc,fill=feature_type))+
#   geom_bar(stat="identity",col="#b30000")+
#   scale_fill_manual(breaks=c("individual_features","all_features"),values=c("#b30000","white"))+
#   facet_wrap(~system_studied)+
#   theme_bw()+
#   geom_hline(yintercept=0.5,linetype="dashed",color="#525252")+
#   ylab("LOOCV RF max accuracy")+
#   ylim(0,1)+
#   theme(legend.position = "none")+
#   coord_flip()
# ggsave(paste(save_path_figures,"fiber_fermented_loocv_rf_host_features.pdf"),width = 4.5,height=5,useDingbats=FALSE)
```

Plot random forest opt variables for host and microbe models
```{r}
all_data_diff_tp7 <- readRDS(paste(save_path_fvf,"all_data_diff_tp7.rds",sep=""))
all_data_diff_tp7_gather <- all_data_diff_tp7 %>% 
  gather(key=Feature,value=diff_tp7,-Participant,-Group)

host_optVariables <- readRDS(paste(save_path_fvf,"host_optVariables.rds",sep="")) %>% 
  dplyr::rename(Feature=optVariables)
microbiome_optVariables <- readRDS(paste(save_path_fvf,"microbiome_optVariables.rds",sep="")) %>% 
  dplyr::rename(Feature=optVariables)

host_optVariables_diff_df <- all_data_diff_tp7_gather %>% 
  filter(Feature %in% host_optVariables$Feature) %>% 
  inner_join(host_optVariables,.)
microbe_optVariables_diff_df <- all_data_diff_tp7_gather %>% 
  filter(Feature %in% microbiome_optVariables$Feature) %>% 
  inner_join(microbiome_optVariables,.)
# ggplot(host_optVariables_diff_df,aes(x=Feature,y=diff_tp7,fill=Group))+
#   geom_boxplot()+
#   facet_wrap(~data_type,scales="free")+
#   coord_flip()+
#   scale_fill_manual(values=c('#bebada','#99d594'))+
#   theme_bw()

#add cazymes to microbe optVariables
cazyme_diff_tp7_optVar <- cazyme_diff_tp7 %>%
  gather(key=Feature,value=diff_tp7,-Participant,-Group,-Timepoint) %>%
  filter(Feature %in% microbiome_optVariables$Feature) %>%
  inner_join(microbiome_optVariables,.) %>%
  na.omit() %>%
  select(-Timepoint)
microbe_optVariables_diff_df <- all_data_diff_tp7_gather %>% 
  filter(Feature %in% filter(microbiome_optVariables,data_type!="cazymes")$Feature) %>% 
  inner_join(microbiome_optVariables,.) %>% 
  na.omit() %>% 
  bind_rows(.,cazyme_diff_tp7_optVar)
# ggplot(microbe_optVariables_diff_df,aes(x=Feature,y=diff_tp7,fill=Group))+
#   geom_boxplot()+
#   facet_wrap(~data_type,scales="free")+
#   coord_flip()+
#   scale_fill_manual(values=c('#bebada','#99d594'))+
#   theme_bw()

host_loocv_results <- readRDS(paste(save_path_fvf,"host_loocv_results.rds",sep="")) 
microbiome_loocv_results <- readRDS(paste(save_path_fvf,"microbiome_loocv_results.rds",sep="")) 

#make supp table showing the variables used in the loocv rf and the group averages
host_optVariables_diff_df_grpAvg <- host_optVariables_diff_df %>% 
  group_by(data_type,rank_order,Feature,Group) %>% 
  dplyr::summarize(diff_tp7_grpAvg=mean(diff_tp7,na.rm = T)) %>% 
  spread(key=Group,value=diff_tp7_grpAvg) %>% 
  right_join(host_loocv_results,.) %>% 
  arrange(-max_acc)
write.csv(host_optVariables_diff_df_grpAvg,paste(save_path_fvf,"host_loocv_opt_variables.csv",sep="")) 

microbe_optVariables_diff_df_grpAvg <- microbe_optVariables_diff_df %>% 
  group_by(data_type,rank_order,Feature,Group) %>% 
  dplyr::summarize(diff_tp7_grpAvg=mean(diff_tp7,na.rm = T)) %>% 
  spread(key=Group,value=diff_tp7_grpAvg) %>% 
  right_join(microbiome_loocv_results,.) %>% 
  arrange(-max_acc)
write.csv(microbe_optVariables_diff_df_grpAvg,paste(save_path_fvf,"microbe_loocv_opt_variables.csv",sep=""))


```



